### PR TITLE
[Spark] Enrich default limits and requests

### PR DESCRIPTION
**Pull request by benbd86**
Added default for sparkjob pods, default for driver and executor limits and requests

Support for ticket [ML-1722](https://jira.iguazeng.com/browse/ML-1722)

### CODE REVIEW

File: mlrun/runtimes/sparkjob/abstract.py [hunk 1]
```diff
@@ -630,33 +654,37 @@ def with_executor_requests(
...
-            generate_resources(mem=mem, cpu=cpu, gpus=gpus, gpu_type=gpu_type),
+            resources,
         )
 
     def with_driver_limits(self, cpu=None):
         """set driver pod cpu limits"""
-        update_in(
-            self.spec.driver_resources, "limits", generate_resources(cpu=str(cpu))
-        )
+        resources = self._verify_spark_job_limits("", cpu)
```
Comment (gtopper) [inline mlrun/runtimes/sparkjob/abstract.py]: As discussed, `""` isn't right. Do we need to add a test to cover this code?

**Review by gtopper — APPROVED** [review]
✅

File: mlrun/runtimes/sparkjob/abstract.py [hunk 2]
```diff
@@ -715,6 +743,48 @@ def spec(self) -> AbstractSparkJobSpec:
...
+        if cpu:
+            verify_field_regex(
+                f"function.spec.{resources_field_name}.limits.cpu",
+                cpu,
+                mlrun.utils.regex.k8s_resource_quantity_regex,
+            )
+        return generate_resources(cpu=str(cpu))
+
+    @staticmethod
+    def _verify_spark_job_requests(
```
Comment (Hedingber) [inline mlrun/runtimes/sparkjob/abstract.py:760]: Why duplicating the logic and not using the existing `_verify_and_set_limits` ?

**Review by Hedingber — CHANGES_REQUESTED** [review]
I have to ask since you didn't write test, did you test it ?

File: mlrun/runtimes/sparkjob/abstract.py (hunk 2 above)
Comment (benbd86) [reply mlrun/runtimes/sparkjob/abstract.py:760]: I've talked to @Tankilevitch about it... it is opposite than in other jobs - requests and limits has different parameters

**Review by Hedingber — APPROVED** [review]

**Pull request merged**